client/pages/post-list.jsx

```diff
@@ -1,50 +1,80 @@
 import React from 'react';
-import { format } from 'date-fns';
+import { format, formatDistance } from 'date-fns';
 
-function Post(props) {
-  const { postTitle, imageUrl, caption, eventDate, endTime, location } = props.post;
-  const date = new Date(eventDate);
-  const endDate = new Date(endTime);
-  const formatDay = format(date, 'do');
-  const formatMonth = format(date, 'LLLL');
-  const formatStartTime = format(endDate, 'hh:mmb');
-  const formatEndTime = format(endDate, 'hh:mmb');
-  const eventDateElement = eventDate
-    ? <div className="event-date">
+class Post extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      clicked: false,
+      comments: this.props.comments
+    };
+    this.toggleEvent = this.toggleEvent.bind(this);
+  }
+
+  toggleEvent() {
+    this.setState({
+      clicked: true
+    });
+  }
+
+  render() {
+    const { postTitle, imageUrl, caption, eventDate, endTime, location, postId } = this.props.post;
+    const date = new Date(eventDate);
+    const endDate = new Date(endTime);
+    const formatDay = format(date, 'do');
+    const formatMonth = format(date, 'LLLL');
+    const formatStartTime = format(date, 'hh:mmb');
+    const formatEndTime = format(endDate, 'hh:mmb');
+    const messages = this.props.comments.map((comment, index) => {
+      const commentDate = new Date(comment.createdAt);
+      const formatComment = formatDistance(new Date(), new Date(commentDate));
+      const match = comment.postId === postId
+        ? <div className="comments row justify-space"><p>{`${comment.userId} : ${comment.content}`}</p>
+        <p className="time-ago">{formatComment}</p></div>
+        : null;
+      return (
+        <div key={index}>{match}</div>
+      );
+    });
+    const memoryComment = !eventDate
+      ? <div className="row add-comment"><button onClick={() => this.props.addComment(postId)}>Add a Comment</button></div>
+      : null;
+    const eventDateElement = eventDate
+      ? <div className="event-date">
         <div className="row justify-center">
             <h1>{formatDay}</h1>
         </div>
         <div className="row justify-center">
             <h3>{formatMonth.substr(0, 3)}</h3>
         </div>
     </div>
-    : null;
-  const eventTimeElement = endTime
-    ? <div className="row">
+      : null;
+    const eventTimeElement = endTime
+      ? <div className="row">
```

Comment:
All of these ternaries are really complicated. Try making an EventPost component and an NewsPost component and render one or the other based on the type of post in your `.map()`.